CMakeLists.txt

```diff
@@ -828,30 +828,40 @@ endif()
 # UNIT TESTING 
 ################################################################
 
```

Comment:
I think this loop only exists to set `UNIT_ID_STR` and `UNIT_ID` and setup CTest. Could these be set from `__FILE__` in the source file instead? That way the whole unit testing build is just another target executable and a small loop to setup CTest.

Reply:
I looked into that, and the issue is that `__FILE__` is a string literal, and we need an identifier specifying the test name, if we want to be able to write 
```
TEST_FUN(ident) { /*...*/}
```
If we'd accept to have 
```
TEST_FUN(ident, { /* ... */ })
```
instead we might be able to do with `__FILE__` only. But it requires patching old tests, and fiddling around with macro expansion .. I don't know if it's worth it.

Reply:
Makes sense to me - never mind, can always be simplified later if we figure out how to do it! Good changes.